src/LinearRemapFV.cpp

```diff
@@ -1315,272 +1314,460 @@ void LinearRemapIntegratedGeneralizedBarycentric(
+												
+						// Contribution of each point to the map
+						for (int i = 0; i < vecContributingFaceI.size(); i++){
+					
+							if( vecContributingFaceWeights[i] < -1e-12 || vecContributingFaceWeights[i] > 1+1e-12 ){
+								std::cout << "\nFound weight value = " << vecContributingFaceWeights[i] << std::endl;
+								_EXCEPTIONT("Non-monotone weight");
 							}
-
+							
+							smatMap(iTargetFace, vecContributingFaceI[i]) +=
+								vecContributingFaceWeights[i]
+								* dQuadPtWeight(k,p)
+								* meshOverlap.vecFaceArea[ixOverlap + j]
+								/ dQuadratureArea
+								/ meshOutput.vecFaceArea[iTargetFace];
+					
 						}
-
-						vecWeights[m] *= vecTriangleAreas[m];
-
-						dWeightTotal += vecWeights[m];
-
-					}
-
-					for (int m = 0; m < iEdges; m++){
-
-						vecWeights[m] /= dWeightTotal;
-
+						
 					}
-
-					double dFaceArea = CalculateFaceArea(meshInputDual.faces[iFaceFinal],meshInputDual.nodes);
-
-					double dsum = 0;
-
-					for (int s = 0; s < iEdges; s++){
-
-						dsum += vecTriangleSubAreas[s];
-
+					
+				}
+				
+			}
+			 
+			// Increment the current overlap index
+			ixOverlap += nOverlapFaces;
+			
+		}
+	
+	}
+	
+	//If the source mesh has holes, use local dual faces
+	
+	else {
+		
+		//Vector of source face centers
+				
+		NodeVector vecSourceCenterArray;	
+		
+		for (int i = 0; i < meshInput.faces.size(); i++){
+	
+			const Face & face = meshInput.faces[i];
+	
+			Node nodeCentroid = GetFaceCentroid(face, meshInput.nodes);
+			
+			nodeCentroid = nodeCentroid.Normalized();
+			
+			vecSourceCenterArray.push_back(nodeCentroid);
+	
+				
+		}
+		
+		// Overlap face index
+		int ixOverlap = 0;
+	
+		// Loop through all source faces
+		for (int ixFirst = 0; ixFirst < meshInput.faces.size(); ixFirst++) {
+	
+			// Output every 1000 overlap elements
+			if (ixFirst % 1000 == 0) {
+				Announce("Element %i/%i", ixFirst, meshInput.faces.size());
+			}
+			
+			// This Face
+			const Face & faceFirst = meshInput.faces[ixFirst];
+			
+			int iEdges = faceFirst.edges.size();
+			
+			//Nodes of the current face
+			
+			NodeVector nodesFaceFirst;
+			
+			for (int i = 0; i < faceFirst.edges.size(); i++){
+				
+				nodesFaceFirst.push_back(meshInput.nodes[faceFirst[i]]);
+				
+			}	
+	
+			// Find the set of Faces that overlap faceFirst
+			int ixOverlapBegin = ixOverlap;
+			
+			int ixOverlapEnd = ixOverlapBegin;
+	
+			for (; ixOverlapEnd < meshOverlap.faces.size(); ixOverlapEnd++) {
+				if (meshOverlap.vecSourceFaceIx[ixOverlapEnd] != ixFirst) {
+					break;
+				}
+			}
+			
+			int nOverlapFaces = ixOverlapEnd - ixOverlapBegin;
+	
+			if( nOverlapFaces == 0 ) continue;
+			
+			std::vector<std::vector<int>> vecContributingFaceI;
+			
+			std::vector<std::vector<double>> vecContributingWeights;		
+	
+			// Loop through all overlap faces associated with this source face
+			for (int j = 0; j < nOverlapFaces; j++) {
```

Comment:
Perhaps a point for later: if bilinear family and barycenter integrated schemes use the same workflow for mesh with and without holes, see opportunities to refactor and unify under a similar structure. If there is a bug, there are a lot of places the same code needs to be modified.

Reply:
Got it.  I will work on unifying the code here and submit in another pull request.